api/handlers.go

```
@@ -43,3 +46,95 @@ func GetTradesByItemHandler(ctx context.Context, itemTradeHistory store.TradeSto
 		}
 	}
 }
+
+func CreateNewAlertHandler(ctx context.Context, alertsDefined store.AlertDefStore) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		//Return a 200 OK response
+		w.WriteHeader(http.StatusOK)
+
+		//Create a var to store the response in
+		var responseConfirmationString string
+		var responseConfirmationStringJSON []byte
+
+		//Create key-value pairs (map) from the URL
+		vars := mux.Vars(r)
+
+		//Obtain from the url which item we are interested in reporting - expected format is item,alertType,priceTrigger
+		newAlertDetails := vars["id"]
+
+		// Split out the new alert details
+		item, alertType, priceTrigger, err := CreateNewAlertIDSplit(newAlertDetails)
+		if err != nil {
+			responseConfirmationString = fmt.Sprintf("There was an error trying to get the item details from the url.  Error: %v", err)
+			responseConfirmationStringJSON, _ = json.Marshal(responseConfirmationString)
+			_, _ = w.Write(responseConfirmationStringJSON)
+			return
+		}
+
+		//Call the function from the service package
+		err = service.CreateNewAlert(ctx, alertsDefined, item, models.AlertValues{AlertType: alertType, PriceTrigger: priceTrigger})
+		//results, err = service.ProcessAlerts(ctx, alertsDefined, itemsToReport)
+		if err != nil {
+			responseConfirmationString = fmt.Sprintf("The HTTP server failed to get the results due to error: %v", err)
+			responseConfirmationStringJSON, _ = json.Marshal(responseConfirmationString)
+			_, _ = w.Write(responseConfirmationStringJSON)
+			return
+		}
+
+		// If the alert was created successfully, return a confirmation message
+		responseConfirmationString = fmt.Sprintf("The alert for item %s has been created successfully.\n The alert type is %d, and the price trigger is %d", item, alertType, priceTrigger)
+		responseConfirmationStringJSON, _ = json.Marshal(responseConfirmationString)
```

don't ignore errors with `_`